galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/FrameworkPropertyFile.java

```diff
@@ -154,6 +154,14 @@ public synchronized List<String> getNamespaces() {
      * @throws FrameworkPropertyFileException
      */
     public synchronized void delete(String key) throws FrameworkPropertyFileException {
+        // Make the current properties as close to the values in the file as 
+        // we can. So when we write they are up-to-date.
+        if (observer!=null) {
```

Comment:
Did you need to null check here? The observer is instantiated by the constructor and will throw an exception if unable to create.

Reply:
OK.

Reply:
Actually, destroy can set the observer to null. So I think we need the checks in case the properties file was destroyed.

Reply:
destroy() sets it to null, and some unit tests try setting properties after the destroy call, so I assumed it was supposed to handle it.